react/tic-tac-toe/src/redux/match/reducer.js

```diff
@@ -26,10 +27,14 @@ function games(state = initialState, action) {
     }
     case actions.JUMP_TO:
       return { ...state, stepNumber: action.payload, xIsNext: action.payload % 2 === 0 };
-    case actions.GET_MATCH_HISTORY:
-      return { ...state, matches: action.payload.data, isLoading: false };
     case actions.SET_LOADING:
       return { ...state, isLoading: action.payload };
+    case actions.GET_MATCHES:
+      return { ...state };
+    case actions.GET_MATCHES_FAILURE:
+      return { ...state, matches: [] };
```

Comment:
You should set the loading to false here too! If not, when the fetch fails, your component will stay loading forever